File: backend/api/schemas/questionario.py
import datetime
from ninja import ModelSchema, Schema
from typing import List

from api.schemas.categoria import CategoriaOut
from .questao import QuestaoOut
from .helpers import IdSchema
from api.models import Questionario


class QuestionarioOut(ModelSchema):
    id: int
    nome: str
    descricao: str
    categoria: CategoriaOut
    questoes: List[QuestaoOut]

    class Config:
        model = Questionario
        model_fields = ['id', 'nome', 'descricao', 'categoria',
                        'questoes', 'created_at', 'updated_at']


class QuestionarioIn(Schema):
    nome: str
    descricao: str
    categoria: IdSchema
    questoes_id: List[int]

File: frontend/src/types/models/Questionario.ts
import { Categoria } from "./Categoria";
import { Questao } from "./Questao";

export interface Questionario {
    id?: number;
    nome: string;
    descricao: string;
    categoria: Categoria;
    questoes: Questao[];
    created_at?: Date;
    updated_at?: Date;
}

export interface QuestionarioPayload {
    id?: number;
    nome: string;
    descricao: string;
    categoria_id: number;
    questoes_id: number[];
}
